language/tools/move-unit-test/src/test_reporter.rs

```diff
@@ -92,20 +92,28 @@ impl TestRunInfo {
 }
 
 impl FailureReason {
-    pub fn no_abort() -> Self {
-        FailureReason::NoAbort("Test did not abort as expected".to_string())
+    pub fn no_error() -> Self {
+        FailureReason::NoError("Test did not fail as expected".to_string())
+    }
+
+    pub fn wrong_error(expected: MoveError, actual: MoveError) -> Self {
+        FailureReason::WrongError(
+            "Test did not fail with the expected error".to_string(),
+            expected,
+            actual,
+        )
```

Comment:
nit: I found the wording in the test cases a little confusing because of the overload of the term "fail" here (i.e. the test failed because it "did not fail as expected") ...maybe I just spent too long thinking about this.  My suggestion is to reserve the "fail" and "pass" as terms to refer to what tests do, and wording around "errors" and producing them  for the code under test, e.g.

- "Test did not error as expected"
- "Test did not produce expected error"

Reply:
...I also remember that we use `expected_failure` as the attribute name, so, that will already make things less clear cut, so I'm less sure about this comment.  Make of it what you will.

Reply:
I like `"Test did not produce expected error"`, thanks!